eslint/eslint-bulk/src/utils/which-eslint.ts

```diff
@@ -0,0 +1,30 @@
+import { execSync } from 'child_process';
+import path from 'path';
+
+const SUPPORTED_VERSIONS = ['v8.6.0', 'v8.7.0', 'v8.21.0', 'v8.22.0', 'v8.23.0', 'v8.23.1'];
+
+export function whichEslint(): string {
+  // Try to find a local ESLint installation
+  try {
+    const localEslintPath = path.dirname(require.resolve('eslint', { paths: [__dirname, process.cwd()] }));
+    const eslintPackageJson = require(path.join(path.dirname(localEslintPath), 'package.json'));
+    const localEslintVersion = eslintPackageJson.version;
+
+    if (SUPPORTED_VERSIONS.includes(localEslintVersion))
+      return `node ${path.join(localEslintPath, 'bin', 'eslint.js')}`;
+  } catch {
+    // If we can't find a local ESLint or read its version, we do nothing and move on to the next step
+  }
+
+  // Try to find a npx ESLint installation
+  try {
+    const npxEslintVersion = execSync('npx eslint -v', { stdio: 'ignore' }).toString().trim();
```

Comment:
`npx` will automatically install some version of `eslint` in a global cache, assuming an NPM registry is reachable at the time.

This seems a bit sloppy:  If we are going to be processing thousands of files and generating suppressions that will be committed to Git, we really should be using the configured version of ESLint for the affected project. If we cannot determine the correct ESLint version, it would be better to fix that problem, than magically proceed with a most likely wrong guess about the ESLint version.

Reply:
This logic has been removed